relay-server/src/utils/pick.rs

```diff
@@ -1,16 +1,47 @@
-/// Returns `true` if `id` is rolled out with a rollout rate `rate`.
+/// Result of a sampling operation, whether to keep or discard something.
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
+pub enum PickResult {
+    /// The item should be kept.
+    #[default]
+    Keep,
+    /// The item should be dropped.
+    Discard,
+}
+
+/// Returns [`PickResult::Keep`] if `id` is rolled out with a rollout rate `rate`.
 ///
 /// Deterministically makes a rollout decision for an id, usually organization id,
 /// and rate.
-pub fn is_rolled_out(id: u64, rate: f32) -> bool {
-    ((id % 100000) as f32 / 100000.0f32) < rate
+pub fn is_rolled_out(id: u64, rate: f32) -> PickResult {
+    match ((id % 100000) as f32 / 100000.0f32) < rate {
+        true => PickResult::Keep,
+        false => PickResult::Discard,
+    }
+}
+
+impl PickResult {
+    /// Returns `true` if the sampling result is [`PickResult::Keep`].
+    pub fn is_keep(self) -> bool {
+        match self {
+            PickResult::Keep => true,
+            PickResult::Discard => false,
+        }
```

Comment:
Maybe with `matches`?
```suggestion
        matches!(self, PickResult::Keep)
```